# BIO 462: Computational Cancer Biology

BIO 462 is a class designed to teach students cancer biology and how 
to use common bioinformatics tools on real cancer datasets. Below are 
links to the Google Colab notebooks for homework assignments 1 through 6:

- [Homework 1: Intro to Cancer Datasets](https://colab.research.google.com/drive/1RV2Sx5cfr9E83afB1eAFiyyo6QF4pEHZ?usp=sharing)
    - Teaches students basic Python Pandas and cptac package functions used to analyze real cancer data presented in Pandas data frames.
      Introduces students to a few of the cancer datasets available in the Python cptac package.


- [Homework 2: Missense Mutation](https://colab.research.google.com/drive/1-9eDUJhc1vLADDgxRick16GtjmSIEu45?usp=sharing)
    - Teaches students how to identify and interpret missense mutations within the cancer datasets. 
      Introduces UniProt data using the UniProt API access.


- [Homework 3: Truncation Mutation](https://colab.research.google.com/drive/1Awq1A5wZV2hEEevCA2phN7NnZnPMiS5X?usp=sharing)
    - Teaches students some analysis techniques for identifying mutations in genes and how those can lead to truncated proteins.
      Introduces Mutplot lollipop plots as visualization of several truncating mutations. 


- [Homework 4: Copy Number Variation](https://colab.research.google.com/drive/1pYGj0MWivX3mevUIZR-a2Sock0fyV4xN?usp=sharing)
    - Teaches students the difference between focal- and arm-level CNV events and how they affect individuals vs populations.
      Introduces line plots as a visualization of the data. 


- [Homework 5: mRNA/Transcriptomics](https://colab.research.google.com/drive/1rfu1Kp8zJfA2rY-65WK1P3HEDkkLKbvu?usp=sharing)
    - Teaches students how to identify differential mRNA abundances through t-testing using a wrap t-test cptac function along with teaching how to perform a pathway enrichment analysis on mRNA and transcriptomics data through GSEA/Enrichr and g:Profiler.
      Describes the protocols for each analysis.
      

- [Homework 6: Proteomics]()
    - Teaches students how to identify differential proteins through t-testing along with how to identify significantly correlated proteins using correlation networks and Pandas correlation functions.
