# BIO 462: Computational Cancer Biology

BIO 462 is a class designed to teach students cancer biology and how to use common bioinformatics tools on real cancer datasets. 
We created this module to provide material for those wishing to teach computational cancer biology or expand their offering of 
bioinformatics topics to undergraduate life science majors. Below are links to the Google Colab notebooks for homework assignments 
1 through 6. Each lesson is designed to contain a weekâ€™s worth of instruction material, so the computational exercises in each 
notebook are estimated to take approximately 6 hours. Nevertheless, we invite and encourage educators to adapt the lessons to best 
fit their specific needs. At the beginning of each Colab notebook, we include a section describing the topic and relevant literature 
to be discussed in the classroom. As the module is completely online, it is appropriate for either traditional in-person instruction 
or online courses.


- [Lesson 1: Introduction to Cancer Datasets](https://colab.research.google.com/drive/1RV2Sx5cfr9E83afB1eAFiyyo6QF4pEHZ?usp=sharing)
    - Introduces students to cancer datasets through the use of Pandas dataframes. 
      Teaches basic dataframe manipulation and cancer clinical data exploration.


- [Lesson 2: Missense Mutation](https://colab.research.google.com/drive/1-9eDUJhc1vLADDgxRick16GtjmSIEu45?usp=sharing)
    - Teaches students identification and interpretation of missense mutations. 
      Integrates protein domain classification with CPTAC data using the UniProt API.


- [Lesson 3: Truncation Mutation](https://colab.research.google.com/drive/1Awq1A5wZV2hEEevCA2phN7NnZnPMiS5X?usp=sharing)
    - Teaches students some analysis techniques for identifying mutations in genes and how those can lead to truncated proteins.
      Introduces extended UniProt integration and visualization of truncating mutations with lollipop plots.


- [Lesson 4: Copy Number Variation](https://colab.research.google.com/drive/1pYGj0MWivX3mevUIZR-a2Sock0fyV4xN?usp=sharing)
    - Teaches students the difference between focal and arm level CNV events and how they affect individuals vs populations.
      Teaches visualization techniques for identifying frequent events in a population.


- [Lesson 5: Transcriptomics](https://colab.research.google.com/drive/1rfu1Kp8zJfA2rY-65WK1P3HEDkkLKbvu?usp=sharing)
    - Teaches students how to identify differential transcripts through t-testing using a wrap t-test function 
      Teaches students how to perform a pathway enrichment analysis on transcriptomics data through GSEA/Enrichr and g:Profiler.
      Describes the protocols for each analysis.
      

- [Lesson 6: Proteomics](https://colab.research.google.com/drive/1qpKBlQ2iPhi6XfoJV2avK4f8CaXn5Nnm?usp=sharing)
    - Teaches students identification of differential proteins and co-expression networks. 
      Introduces advanced UniProt integration.
