# BIO 462: Computational Cancer Biology

BIO 462 is a class designed to teach students cancer biology and how 
to use common bioinformatics tools on real cancer datasets. Below are 
links to the Google Colab notebooks for homework assignments 1 through 6:

- [Homework 1: Coding check](https://colab.research.google.com/drive/1RV2Sx5cfr9E83afB1eAFiyyo6QF4pEHZ?usp=sharing)
    - Teaches basic python Pandas data frame operations.


- [Homework 2: Missense mutation](https://colab.research.google.com/drive/1-9eDUJhc1vLADDgxRick16GtjmSIEu45?usp=sharing)
    - Teaches students how to identify and interpret missense
      mutations in cancer datasets. Introduces the Uniprot API.


- [Homework 3: Truncation mutation](https://colab.research.google.com/drive/1Awq1A5wZV2hEEevCA2phN7NnZnPMiS5X?usp=sharing)
    - Teaches students some analysis techniques for identifying
      mutations in genes and how those can lead to truncated proteins.
      Introduces the Mutplot lollipop plot tool.


- [Homework 4: Copy number mutation](https://colab.research.google.com/drive/1pYGj0MWivX3mevUIZR-a2Sock0fyV4xN?usp=sharing)
    - Teaches students the difference between focal and arm level
      CNV events and how they affect individuals vs populations.
      Introduces line plots as a visualization of the data. 


- [Homework 5: mRNA]()
    - Teaches students how to perform a pathway enrichment analysis
      on mRNA and transcriptomics data through GSEA and g:Profiler.
      Describes the protocols for each analysis. Introduces a
      built-in wrapper t-test from the cptac API.
      

- [Homework 6: Proteomics]()
    - Teaches students how to identify differential proteins through
      t-testing along with how to identify significantly correlated
      proteins. Introduces correlation tools to create data frames
      and correlation networks.
